### PR TITLE
Fix repl script mode

### PR DESCRIPTION
**Pull request by elprans**
Unbreak the "script from stdin" mode: fix handling of password prompt,
as well as exception rendering and exit code.

### CODE REVIEW

File: tests/test_cli.py [hunk 1]
```diff
@@ -72,3 +72,7 @@ class TestCLI(tb.ConnectedTestCase, tb.CLITestCaseMixin):
         result = self.run_cli('create', 'role', 'foo', '--allow-login',
                               '--password', input='foo-pass\n')
         self.assertIn('input is not a TTY', result.output)
+
+    async def test_cli_repl_script(self):
```
Comment (1st1) [inline tests/test_cli.py]: I'd also test that an edgeql error causes a non-zero return code.
